test/unit/StakingPool/calculatePremium.js

```diff
@@ -0,0 +1,480 @@
+    const { stakingPool, cover } = this;
+    const { PRICE_CHANGE_PER_DAY, INITIAL_PRICE_DENOMINATOR } = this.config;
+    const initialPrice = coverProductTemplate.initialPriceRatio;
+    const daysToMove = 1;
+    const expectedPrice = initialPrice - PRICE_CHANGE_PER_DAY * daysToMove;
+    await setNextBlockDaysForward(daysToMove);
+    await cover.allocateCapacity({ ...buyCoverParamsTemplate }, coverId, stakingPool.address);
+    const expectedPremium = buyCoverParamsTemplate.amount
+      .mul(expectedPrice)
+      .div(INITIAL_PRICE_DENOMINATOR)
+      .div(periodsInYear);
+    expect(await cover.lastPremium()).to.be.equal(expectedPremium);
+    {
+      const product = await stakingPool.products(productId);
+      const daysToMove = 100;
+      await setNextBlockDaysForward(daysToMove);
+      await cover.allocateCapacity({ ...buyCoverParamsTemplate }, coverId, stakingPool.address);
+      const expectedPremium = buyCoverParamsTemplate.amount
+        .mul(product.targetPrice)
+        .div(INITIAL_PRICE_DENOMINATOR)
+        .div(periodsInYear);
+      expect(await cover.lastPremium()).to.be.equal(expectedPremium);
+    }
+  });
+
+  it('should correctly calculate price when all coverage is bought in a single purchase', async function () {
+    const { stakingPool, cover } = this;
+    const [coverBuyer] = this.accounts.members;
+    const { GLOBAL_CAPACITY_RATIO, INITIAL_PRICE_DENOMINATOR, PRICE_CHANGE_PER_DAY, NXM_PER_ALLOCATION_UNIT } =
+      this.config;
+    const amount = stakedNxmAmount.mul(2 /* capacityRatio */);
+    const buyCoverParams = { ...buyCoverParamsTemplate, amount };
+    const { totalCapacity } = await stakingPool.getActiveTrancheCapacities(
+      buyCoverParamsTemplate.productId,
+      GLOBAL_CAPACITY_RATIO,
+      coverProductTemplate.capacityReductionRatio,
+    );
+    const product = await stakingPool.products(productId);
+    // calculate premiums
+    const basePrice = await calculateBasePrice(stakingPool, product, PRICE_CHANGE_PER_DAY);
+    const basePremium = amount.mul(basePrice).div(INITIAL_PRICE_DENOMINATOR);
+    const { surgePremium, surgePremiumSkipped } = await calculateSurgePremium(
+      stakingPool,
+      amount,
+      0 /* activeAllocations */,
+      totalCapacity,
+      this.config,
+    );
+    const { timestamp } = await ethers.provider.getBlock('latest');
+    const calculatedPremium = await stakingPool.calculatePremium(
+      product,
+      buyCoverParams.period,
+      amount.div(NXM_PER_ALLOCATION_UNIT), // cover amount
+      0, // initialCapacityUsed
+      totalCapacity,
+      product.targetPrice, // targetPrice
+      timestamp, // current block timestamp
+    );
+    const expectedPremium = basePremium.add(surgePremium).sub(surgePremiumSkipped).div(periodsInYear);
+    expect(calculatedPremium.premium).to.be.equal(expectedPremium);
+    await cover.connect(coverBuyer).allocateCapacity(buyCoverParams, coverId, stakingPool.address);
```

Comment:
The tests shouldn't have an entry point with `allocateCapacity` even for the `requestAllocation` because they are unit tests.

This looks more like an integration test, even though it isn't.

All entry points should be through StakingPool methods

I would have another file that tests `requestAllocation`. 

But againt calculatePremium tests should *only* test function with the same name.